src/metabase/api/notification.clj

```diff
@@ -9,12 +9,36 @@
 
 (set! *warn-on-reflection* true)
 
+(defn- get-notification
+  [id]
+  (some-> (t2/select-one :model/Notification id)
+          models.notification/hydrate-notification))
+
 (api/defendpoint GET "/:id"
   "Get a notification by id."
   [id]
   {id  ms/PositiveInt}
-  (-> (t2/select-one :model/Notification id)
-      api/check-404
-      models.notification/hydrate-notification))
+  (api/check-404 (get-notification id)))
+
+(api/defendpoint POST "/"
+  "Create a new notification, return the created notification."
+  [:as {body :body}]
+  {body models.notification/FullyHydratedNotification}
+  (models.notification/hydrate-notification
+   (models.notification/create-notification!
+    (dissoc body :handlers :subscriptions)
```

Comment:
the fact that `FullyHydratedNotification` is not closed (and maybe it should be) gives me some feeling of uneasiness just passing whatever here. :) TBH I'd rather see `select-keys` than `dissoc` in places like that, or _at least_ a closed map.

Reply:
yea I dislike it so much, but last I checked in malli if you have a closed map, and it's merge with another map or dispatch it no longer closed.

I plan to revisit this and see if we can have a workaround, also I plan to use convert to use registry so we don't have to re-eval all methods when we change one schema.